Sources/Core/GraphRequestConnection.swift

```diff
@@ -152,4 +150,134 @@ class GraphRequestConnection: GraphRequestConnecting {
     }
     return URLRequest(url: url)
   }
+
+  private func urlRequest(with graphRequest: GraphRequest) -> URLRequest {
+    guard let url = URLBuilder().buildURL(path: graphRequest.graphPath.description) else {
+      fatalError("Should never fail to build a url from the url builder")
+    }
+
+    return URLRequest(url: url)
+  }
+
+  /**
+   The recommended way of retrieving objects from the Graph API
+
+   - Parameter remoteType: A generic Decodable type to return.
+   It is recommended that you keep this type flexible and use it as a starting point for building stronger typed
+   (canonical) models for use in your application
+   - Parameter graphRequest: The graph request object to use in creating the data request
+   - Parameter completion: A Result type with a Success of the specified Decodable type and a Failure of Error
+
+   - Returns
+   URLSessionTaskProxy - a wrapper for a url session task that allows you to cancel an in-flight
+   request
+   */
+  func getObject<RemoteType: Decodable>(
+    ofType remoteType: RemoteType.Type,
+    for graphRequest: GraphRequest,
+    completion: @escaping (Result<RemoteType, Error>) -> Void
```

Comment:
Type alias?

Reply:
Sounds good to me.

Reply:
I did it for the other completion but based in it being generic I don't think we gain much by trying to typealias this. Ex:

    typealias GetRemoteTypeCompletion<RemoteType> = (Result<RemoteType, Error>) -> Void

Maybe I'm doing this inefficiently though. Open to suggestions.